source/Irrlicht/CIrrDeviceSDL.cpp

```diff
@@ -648,6 +580,19 @@ bool CIrrDeviceSDL::run()
 			}
 			break;
 
+		case SDL_TEXTINPUT:
+			{
+				irrevent.EventType = irr::EET_STRING_INPUT_EVENT;
+				size_t size = strlen(SDL_event.text.text);
+				wchar_t tmp[size*2];
+				mbstowcs(tmp, SDL_event.text.text, size*2);

```

Comment:
there's a wrapper function for this btw
https://github.com/minetest/irrlicht/blob/master/include/irrString.h#L1431